ggml/CMakeLists.txt

```diff
@@ -172,6 +172,11 @@ set   (GGML_SYCL_TARGET "INTEL" CACHE STRING
 set   (GGML_SYCL_DEVICE_ARCH "" CACHE STRING
                                             "ggml: sycl device architecture")
 
+option(GGML_OPENCL                          "ggml: use OpenCL"                                OFF)
+option(GGML_OPENCL_PROFILING                "ggml: use OpenCL profiling (increases overhead)" OFF)
+option(GGML_OPENCL_EMBED_KERNELS            "ggml: embed kernels"                             ON)
+option(GGML_OPENCL_USE_ADRENO_KERNELS       "ggml: use optimized kernels for Adreno"          ON)
```

Comment:
It would be nice if this could be detected automatically at runtime instead.

Reply:
Not really possible for cross-compilation.
ie building with Android NDK, or building with MinGW (Linux -> Windows)

Reply:
Can you help me understand why this would not be possible? I was accidentally building with `GGML_OPENCL_USE_ADRENO_KERNELS` enabled, and the build succeeded without issues. The kernels failed to compile, but that can be detected at runtime.

Reply:
Ha. I just realized that misunderstood your original suggestion. :) 
I interpreted it as "autodetect Adreno at build time". My bad. You did say _runtime_.
Yes, in theory, we should be able to do that. i.e. Include all kernels at build time and then at runtime decide which kernels to use based on what GPU we find. We'll need to think about that. 
Might be better to do what you what you did with the x86-64 CPU backend (and what we'll do with the ARM64 flavors), which is to build several flavors of `opencl-adreno`, `opencl-intel`, etc to avoid lots of checks at runtime.

Reply:
That could work as well, but that would also require implementing a score function to determine which one to load, and that may be not much simpler than doing the same check when loading the kernels.

Reply:
Well, if we do `opencl-adreno`, `opencl-intel`, ..., `opencl-generic` then we probably don't need the score because they can just ignore unsupported GPUs. i.e. adreno would only claim Adreno GPU, etc and generic would in theory run everywhere.